syntax = "proto3";

message ScreenCoordinates {
  float x = 1;
  float y = 2;
}

message GazeSample {
    // Timestamp from device's internal clock in microseconds.
    int64 device_timestamp_us = 1;

    // Timestamp from host's monotonic clock in microseconds.
    int64 system_timestamp_us = 2;

    optional ScreenCoordinates left_eye = 3;
    optional ScreenCoordinates right_eye = 4;
}

message GazeBundle {
    repeated GazeSample samples = 1;
}